File: frontend/src/microservices/operations/overall/objects/ManagerTaskResponse.ts
// 本文件定义 `ListManagerTasksPlanner` 返回的单条任务数据。

import type { SupplierReviewDecisionResponse } from './SupplierReviewDecisionResponse'

export type ManagerTaskResponse = {
  taskId: string | null
  taskType: string
  orderItemId: string
  orderId: string
  orderItemKind: string
  detailLabel: string
  supplierReviewStatus: string
  supplierReviewDecision: SupplierReviewDecisionResponse | null
  summaryLabel: string
  bookedAmount: string
  bookedCurrency: string
  requestedAt: string
  createdAt: string
  reviewedAt: string | null
  reviewedBy: string | null
  reviewNote: string | null
}

export const managerTaskResponseFromJson = (json: string): ManagerTaskResponse =>
  JSON.parse(json) as ManagerTaskResponse

export const managerTaskResponseToJson = (value: ManagerTaskResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerTaskResponse.scala
// 本文件定义 `ListManagerTasksPlanner` 返回的单条任务数据。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerTaskResponse(
    taskId: Option[String],
    taskType: String,
    orderItemId: String,
    orderId: String,
    orderItemKind: String,
    detailLabel: String,
    supplierReviewStatus: String,
    supplierReviewDecision: Option[SupplierReviewDecisionResponse],
    summaryLabel: String,
    bookedAmount: String,
    bookedCurrency: String,
    requestedAt: String,
    createdAt: String,
    reviewedAt: Option[String],
    reviewedBy: Option[String],
    reviewNote: Option[String]
)
object ManagerTaskResponse:
  given sourceEncoder: Encoder[ManagerTaskResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerTaskResponse] = deriveDecoder
